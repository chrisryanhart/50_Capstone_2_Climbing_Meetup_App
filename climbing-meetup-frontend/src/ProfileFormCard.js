import React, {useState,useContext} from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
// import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
// import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import CountContext from './UserContext';
import Button from '@material-ui/core';

const useStyles = makeStyles({
  root: {
    minWidth: 275,
  },
  bullet: {
    display: 'inline-block',
    margin: '0 2px',
    transform: 'scale(0.8)',
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
});

export default function ProfileFormCard() {
  const classes = useStyles();
  // const bull = <span className={classes.bullet}>â€¢</span>;
  const {registerUser} = useContext(CountContext);

  const INITIAL_STATE = {
    username:'',
    password:'',
    name:'',
    imageUrl:'',
    age:'',
    gender:'',
    isParent:'',
    location:'',
    bio:''
  }

  const [newProfileFormData, setNewProfileFormData] = useState(INITIAL_STATE);

  const handleChange = (e) => {
    e.preventDefault();
    const {value, name} = e.target;

    setNewProfileFormData(data => ({...data, [name]: value}));
  }

  const handleSubmit = (e) => {

  }

  return (
    <Card className={classes.root}>
      <CardContent>
        <Typography variant="h5" component="h2">
          Complete New Profile Form:
        </Typography>

        <Typography variant="body2" component="div">
            <form>
                <div>
                    <label for='username'>Username: </label>
                    <input onChange={handleChange} name='username' value={newProfileFormData.username}/>
                </div>
                <div>
                    <label for='password'>Password: </label>
                    <input onChange={handleChange} name='password' value={newProfileFormData.password}/>  
                </div>
                <div>
                    <label for='name'>Name: </label>
                    <input onChange={handleChange} name='name' value={newProfileFormData.name}/>  
                </div>
                <div>
                    <label for='imageUrl'>Profile Image: </label>
                    <input onChange={handleChange} name='imageUrl' value={newProfileFormData.imageUrl}/>  
                </div>
                <div>
                    <label for='age'>Age: </label>
                    <input onChange={handleChange} name='age' value={newProfileFormData.age}/>  
                </div>
                <div>
                    <label for='isParent'>Parent: </label>
                    <input onChange={handleChange} name='isParent' value={newProfileFormData.isParent}/>  
                </div>
                <div>
                    <label for='location'>Location: </label>
                    <input onChange={handleChange} name='location' type="number" value={newProfileFormData.location}/>  
                </div>
                <div>
                    <label for='bio'>Bio: </label>
                    <textarea onChange={handleChange} name='bio' value={newProfileFormData.bio}></textarea>  
                </div>
                <Button onClick={handleSubmit} variant='contained'>Submit</Button>
            </form>
        </Typography>
      </CardContent>

    </Card>
  );
}
