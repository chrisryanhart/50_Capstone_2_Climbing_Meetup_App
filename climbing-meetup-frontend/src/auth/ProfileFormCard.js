import React, {useState,useContext} from 'react';
import { useHistory } from 'react-router-dom';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
// import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
// import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import CountContext from '../UserContext';
import Button from '@material-ui/core/Button';

const useStyles = makeStyles({
  root: {
    minWidth: 275,
  },
  bullet: {
    display: 'inline-block',
    margin: '0 2px',
    transform: 'scale(0.8)',
  },
  title: {
    fontSize: 14,
  },
  pos: {
    marginBottom: 12,
  },
});

export default function ProfileFormCard() {
  const classes = useStyles();
  // const bull = <span className={classes.bullet}>â€¢</span>;
  const {registerUser} = useContext(CountContext);

  const history = useHistory();

  const INITIAL_STATE = {
    username:'',
    password:'',
    name:'',
    profile_image:'',
    user_age: null,
    user_gender: '',
    is_parent: false,
    location_id: 466,
    bio:''
  }

  const [hasError, sethasError] = useState(false);
  const [errorMessage,setErrorMessage] = useState('');
  const [newProfileFormData, setNewProfileFormData] = useState(INITIAL_STATE);

  const handleCheckboxChange = (e) => {
    // e.preventDefault();
    const value = e.target.checked;
    const name = e.target.name;

    setNewProfileFormData(data => ({...data,[name]:value}));

  }

  const handleGenderChange = (e) => {
    const {value} = e.target;

    setNewProfileFormData(data => ({...data, 'user_gender': value}));
  }


  const handleChange = (e) => {
    // e.preventDefault();
    const {value, name} = e.target;

    setNewProfileFormData(data => ({...data, [name]: value}));
  }

  const handleSubmit = async (e) => {
    e.preventDefault();
    if(newProfileFormData.user_age === ''){
      sethasError(true);
      setErrorMessage('Age input required');
      return;
    }
    let res = await registerUser(newProfileFormData);

    // if type error key doesn't exist, registration was successful
    if(typeof(res.error)==='undefined'){
      setNewProfileFormData(INITIAL_STATE);
      history.push(`/users/${res}`);
    }else{
      // if error exists
      sethasError(true);
      setErrorMessage(res.error.message);
    }
  }
  console.log(newProfileFormData.user_gender);

  return (
    <Card className={classes.root}>
      <CardContent>
        <Typography variant="h5" component="h2">
          Complete New Profile Form:
        </Typography>

        {hasError && <Typography style={{backgroundColor:'red'}}  variant="h6" component="h2">
            <p style={{color:'white'}}><b>Error: {errorMessage}</b></p>
        </Typography>}

        <Typography variant="body2" component="div">
            <form>
                <div>
                    <label htmlFor='username'>Username: </label>
                    <input onChange={handleChange} name='username' value={newProfileFormData.username}/>
                </div>
                <div>
                    <label htmlFor='password'>Password: </label>
                    <input type="password" onChange={handleChange} name='password' value={newProfileFormData.password}/>  
                </div>
                <div>
                    <label htmlFor='name'>Name: </label>
                    <input onChange={handleChange} name='name' value={newProfileFormData.name}/>  
                </div>

                    <label htmlFor='user_age'>Age: </label>
                    <input type="number" onChange={handleChange} name='user_age' value={newProfileFormData.user_age}/>  
                <div> 
                    <label htmlFor='is_parent'>Parent: </label>
                    <input type="checkbox" checked={newProfileFormData.is_parent} id='is_parent' onChange={handleCheckboxChange} name='is_parent' />  

                </div>
                <div>
                    <label>Gender: </label>
                    <input type="radio" onChange={handleGenderChange} name='user_gender' value="male"/>  
                    <label className='radio' htmlFor='user_gender'>Male </label>
                    <input type="radio" onChange={handleGenderChange} name='user_gender' value="female"/>  
                    <label className='radio' htmlFor='user_gender'>Female </label>
                    <input type="radio" onChange={handleGenderChange} name='user_gender' value="nonbinary"/>  
                    <label className='radio' htmlFor='user_gender'>Non-Binary </label>  
                </div>

                <div>
                    <label htmlFor='bio'>Bio: </label>
                    <textarea onChange={handleChange} name='bio' value={newProfileFormData.bio}></textarea>  
                </div>
                <Button type='submit' onClick={handleSubmit} variant='contained'>Submit</Button>
            </form>
        </Typography>
      </CardContent>

    </Card>
  );
}
