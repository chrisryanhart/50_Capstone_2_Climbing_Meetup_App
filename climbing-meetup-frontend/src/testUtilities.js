import React from "react";
import UserContext from "./UserContext";

// const demoUser = {
//   username: "testuser",
//   first_name: "testfirst",
//   last_name: "testlast",
//   email: "test@test.net",
//   photo_url: null,
// };

// token
const userToken = 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwidXNlcm5hbWUiOiJ0ZXN0dXNlcjEiLCJpYXQiOjE2NzQ0MzYyMDZ9.m7tigic9I4D58MYFqDDBmc7cnZTSB5qtLkij5-Ol36A';

const regMock = jest.fn();

const UserProvider =
    ({ children, currUserId = 1, token=userToken, registerUser={regMock}}) => (
    <UserContext.Provider value={{ currUserId, token, registerUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider,regMock };