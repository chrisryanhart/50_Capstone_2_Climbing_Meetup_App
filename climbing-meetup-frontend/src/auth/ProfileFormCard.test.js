import React from "react";
import { render, fireEvent } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
import ProfileFormCard from "./ProfileFormCard";
import { UserProvider, regMock } from "../testUtilities";
import { MemoryRouter } from "react-router-dom";




it("renders without crashing", function() {

  render(
    <MemoryRouter>
        <UserProvider>
            <ProfileFormCard />
        </UserProvider>
    </MemoryRouter>

    );
});


it("submits form when submit button pushed", function() {
    const registerMock = jest.fn()

    const {getByText, getByLabelText } = render(
      <MemoryRouter>
        <UserProvider registerUser={registerMock} >
            <ProfileFormCard />
        </UserProvider>
      </MemoryRouter>
      );
    
    const formTitle = getByText('Complete New Profile Form:');
    // expect(submitFormMock)

    expect(formTitle).toBeInTheDocument();

    const submitButton = getByText('Submit');

    fireEvent.click(submitButton);

    // expect(regMock).toHaveBeenCalled();

    expect(registerMock).toHaveBeenCalled();

    // console.log('test');

    // expect(getByText('New Meetup Form')).toBeInTheDocument();
    
});