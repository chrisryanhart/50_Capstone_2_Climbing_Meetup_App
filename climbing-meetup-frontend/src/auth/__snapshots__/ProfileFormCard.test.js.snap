// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`detects snapshot changes 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-5 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5"
      >
        Complete New Profile Form:
      </h2>
      <div
        class="MuiTypography-root MuiTypography-body2"
      >
        <form>
          <div>
            <label
              for="username"
            >
              Username: 
            </label>
            <input
              name="username"
              value=""
            />
          </div>
          <div>
            <label
              for="password"
            >
              Password: 
            </label>
            <input
              name="password"
              type="password"
              value=""
            />
          </div>
          <div>
            <label
              for="name"
            >
              Name: 
            </label>
            <input
              name="name"
              value=""
            />
          </div>
          <label
            for="user_age"
          >
            Age: 
          </label>
          <input
            name="user_age"
            type="number"
            value=""
          />
          <div>
            <label
              for="is_parent"
            >
              Parent: 
            </label>
            <input
              id="is_parent"
              name="is_parent"
              type="checkbox"
            />
          </div>
          <div>
            <label>
              Gender: 
            </label>
            <input
              name="user_gender"
              type="radio"
              value="male"
            />
            <label
              class="radio"
              for="user_gender"
            >
              Male 
            </label>
            <input
              name="user_gender"
              type="radio"
              value="female"
            />
            <label
              class="radio"
              for="user_gender"
            >
              Female 
            </label>
            <input
              name="user_gender"
              type="radio"
              value="nonbinary"
            />
            <label
              class="radio"
              for="user_gender"
            >
              Non-Binary 
            </label>
          </div>
          <div>
            <label
              for="bio"
            >
              Bio: 
            </label>
            <textarea
              name="bio"
            />
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
